app/src/components/common/SharingModal/ShareLinkView.tsx

```diff
@@ -0,0 +1,246 @@
+              {email}
+              <span title="Remove" data-tag-handle onClick={() => removeEmail(index)}>
+                <img alt="remove" src="/assets/img/workspaces/cross.svg" />
+              </span>
+            </div>
+          )}
+        />
+      </div>
+    ),
+    [permittedEmailsList]
+  );
+
+  const handleAddRecipient = (recipients: string[]) => {
+    setPermittedEmailsList(recipients);
+    const newRecipients = recipients.map((recipient) => ({
+      label: recipient,
+      value: recipient,
+    }));
+    setSharedListRecipients(newRecipients);
+  };
+
+  const validateSharedListName = useCallback(() => {
+    //eslint-disable-next-line
+    const specialCharacters = /[`!@#$%^&*()_+\=\[\]{};':"\\|,.<>\/?~]/;
+    return specialCharacters.test(sharedListName);
+  }, [sharedListName]);
+
+  const handleSharedListCreation = useCallback(() => {
+    const isSharedListNameNotValid = validateSharedListName();
+    if (!sharedListName) {
+      setError("Shared list name cannot be empty");
+      return;
+    }
+    if (isSharedListNameNotValid) {
+      setError("Shared list name cannot have special characters");
+      return;
+    }
+    setIsMailSent(false);
+    setShareableLinkData(null);
+    setError(null);
+
+    try {
+      setIsLinkGenerating(true);
+      createSharedList(
+        appMode,
+        rulesToShare,
+        sharedListName,
+        groupwiseRulesToPopulate,
+        sharedLinkVisibility,
+        sharedListRecipients,
+        user?.details?.profile?.uid
+      ).then(({ sharedListId, sharedListName, sharedListData }) => {
+        if (sharedLinkVisibility === SharedLinkVisibility.PRIVATE && sharedListRecipients.length) {
+          sendSharedListShareEmail({
+            sharedListData: sharedListData,
+            recipientEmails: sharedListRecipients,
+          })
+            .then((res: any) => {
+              if (res.data.success) setIsMailSent(true);
+            })
+            .catch((err) => {
+              Logger.log("send shared list email failed : ", err);
+              toast.error("Opps! Couldn't send the notification");
+            });
+        }
+        setShareableLinkData({
+          link: getSharedListURL(sharedListId, sharedListName),
+          visibility: sharedLinkVisibility,
+        });
+        setIsLinkGenerating(false);
+      });
+    } catch (e) {
+      setIsLinkGenerating(false);
+    }
+  }, [
+    appMode,
+    rulesToShare,
+    sharedListName,
+    groupwiseRulesToPopulate,
+    sharedLinkVisibility,
+    sharedListRecipients,
+    user?.details?.profile?.uid,
+    sendSharedListShareEmail,
+    validateSharedListName,
+  ]);
+
+  useEffect(() => {
+    if (!rulesToShare?.length) return;
+
+    if (rulesToShare.length > 1)
+      setSharedListName(`requesly_shared_list_${epochToDateAndTimeString(new Date()).split(" ")[0]}`);
+    else setSharedListName(singleRuleData?.name);
+  }, [rulesToShare, singleRuleData]);
```

Comment:
In `useEffect`'s callback we are only using the `rulesToShare.length` and `singleRuleData?.name` so lets keep that only in the dependency